### DISCUSSION TITLE
Clarify how to achieve anchor wrap without Header content duplication

### DISCUSSION THREAD

**Discussion by jhogue** [allejo/jekyll-anchor-headings#16] [Q&A] [answered]
**My Markdown**

```markdown
### History

The Atlantic Coal Company “Blue Coal”…
```

**Snippet Usage**

```liquid
{% 
  include content_anchors.html 
    html=property.content 
    beforeHeading=true 
    anchorTitle="Press to bookmark this heading" 
    anchorBody="<span>#</span>%heading%" 
%}
```

**Expected HTML**

```html
<h3 id="reason-for-demolition">
    <a href="#reason-for-demolition"><span>#</span>Reason For Demolition</a>
</h3>
<p>The Atlantic Coal Company “Blue Coal”…
```

**Actual HTML**

```html
<h3 id="reason-for-demolition">
    <a href="#reason-for-demolition"><span>#</span>Reason For Demolition</a> Reason For Demolition
</h3>
<p>The Atlantic Coal Company “Blue Coal”…
```

**Notes**

Sorry if I'm dense. This seems to work very well, I just don't understand how to use it perhaps. I'd like the anchor to wrap around the entire heading string, but as you see, when I specify to use the 
`%heading%` placeholder, the string is duplicated. Please advise.

**Comment by allejo** [comment, marked as answer]
This project is intended to prepend or append a custom anchor to every heading within `content`. It's not designed to wrap a heading and make it clickable. However, if that's what you'd like to do, modify this bit of code in your site:

https://github.com/allejo/jekyll-anchor-headings/blob/689656c554af533267b35554ce79b36cc86e54c7/_includes/anchor_headings.html#L96-L106

Change it something like:

```liquid
     {% capture new_heading %} 
       <h{{ _hAttrToStrip }} 
         {{ include.bodyPrefix }} 
         {{ anchor }} 
         {{ include.bodySuffix }} 
       </h{{ _workspace | last }} 
     {% endcapture %} 
```

This way, it'll completely drop the heading's content and just use the anchor's body.

**Comment by jhogue** [comment]
Oh I see, thank you for the late night request. I was trying to make the touch target to get the anchor appended to the URL string for sharing larger. I see now what the intention is. I could modify the Liquid as you have suggested or I could use CSS to make the smaller anchor carry absolute position, effectively making it larger that way. I'll weigh those options. Thank you.
